spec.emu

```diff
@@ -271,6 +463,86 @@ contributors: Luca Casonato, Guy Bedford
             </tr>
           </table>
         </emu-table>
+
+        <emu-clause id="sec-LoadRequestedModules" type="concrete method">
+          <h1>
+            LoadRequestedModules (
+              optional _hostDefined_: anything,
+            ): a Promise object
+          </h1>
+          <dl class="header">
+            <dt>for</dt>
+            <dd>a Cyclic Module Record _module_</dd>
+          </dl>
+
+          <emu-clause id="sec-InnerModuleLoading" type="abstract operation">
+            <h1>
+              InnerModuleLoading (
+                _state_: a GraphLoadingState Record,
+                _module_: a Module Record,
+                <ins>optional _referrer_: a Cyclic Module Record,</ins>
+                <ins>optional _specifier_: a String,</ins>
+              ): ~unused~
+            </h1>
+            <dl class="header">
+              <dt>description</dt>
+              <dd>It is used by LoadRequestedModules to recursively perform the actual loading process for _module_'s dependency graph.</dd>
+            </dl>
+
+            <emu-alg>
+              1. Assert: _state_.[[IsLoading]] is *true*.
+              1. <ins>Let _importOnlyAsReflection_ be *false*.</ins>
+              1. <ins>If _referrer_ has been specified, then</ins>
+                1. <ins>Assert: _specifier_ has been specified.</ins>
+                1. <ins>If _referrer_.[[RequestedModuleReflections]] contains _specifier_, set _importOnlyAsReflection_ to *true*.</ins>
+              1. If <ins>_importOnlyAsReflection_ is *false*,</ins> _module_ is a Cyclic Module Record, _module_.[[Status]] is ~new~, and _state_.[[Visited]] does not contain _module_, then
+                1. Append _module_ to _state_.[[Visited]].
+                1. Let _requestedModulesCount_ be the length of _module_.[[RequestedModules]].
+                1. Set _state_.[[PendingModulesCount]] to _state_.[[PendingModulesCount]] + _requestedModulesCount_.
+                1. <ins>Let _allRequestedModules_ be the list-concatenation of _module_.[[RequestedModules]] and _module_.[[RequestedModuleReflections]].</ins>
```

Comment:
Rather than iterating `_allRequestedModules_` we could have two separate iterations, and then set a boolean argument like `reflection` on `InnerModuleLoading`? Alternatively we could even have an `InnerModuleLoadingReflection` to avoid the nested branching.